contracts/interfaces/IRepaymentController.sol

```diff
@@ -5,9 +5,9 @@ pragma solidity 0.8.18;
 interface IRepaymentController {
     // ============== Lifeycle Operations ==============
 
-    function repay(uint256 loanId) external;
+    function repay(uint256 loanId, uint256 amount) external;
```

Comment:
I added explicit routes to the RC for 'full' repayments. Even though yes correct about the max unit256, it seems like an anti pattern and could save a little gas since the amount checks are not needed in full repayment. Added tests for this change to the `PartialRepayments.ts` file.

For tests showing repayments where the borrower specifies and amount that is larger than the amount due, see the RepaymentController.ts and Rollover.ts files